### PR TITLE
refactor(frontend): simplify bind_columns_from_source

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

motivation: to make code--

Can review commit by commit, so the correctness might be more easy to verify.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/frontend/src/handler/create_source.rs [hunk 1]
```diff
@@ -344,93 +328,58 @@ pub(crate) async fn bind_columns_from_source(
...
+            stream_source_info.row_schema_location = protobuf_schema.row_schema_location.0.clone();
+            stream_source_info.proto_message_name = protobuf_schema.message_name.0.clone();
+            stream_source_info.key_message_name =
+                get_key_message_name(&mut format_encode_options_to_consume);
+            stream_source_info.name_strategy =
+                name_strategy.unwrap_or(PbSchemaRegistryNameStrategy::Unspecified as i32);
+
+            Some(
+                extract_protobuf_table_schema(
+                    &protobuf_schema,
```
Comment (xxchan) [inline src/frontend/src/handler/create_source.rs:341]: I believe protobuf and avro can also reuse a lot of code, however `extract_protobuf_table_schema` looks strange and I'm not brave enough to change it in this PR, which contains only equivalent rewrites. 🤡

**Comment by xxchan** [comment]
Oops

**Comment by xxchan** [comment]
Why nexmark-backwards-compact failed? It should not be related with this. 🤡

image

**Comment by xxchan** [comment]
For main-cron, only `background_ddl recovery test (deterministic simulation)` failed, but it seems also to fail in daily pipeline, so I will ignore this..

**Comment by xxchan** [comment]
BTW, Backwards compatibility test failed in PR pipeline, but succeeded in main-cron pipeline...

**Comment by xxchan** [comment]
```diff
-   1000	1001	10465998	2015-07-15 00:00:22.005
+   1000 1001 10465998 2015-07-15 00:00:22.005
+   1000 1001 10465998 2015-07-15 00:00:22.005
```
It seems it's `delete` not handled correctly.

**Comment by xxchan** [comment]
It seems `include ./delete.slt.part` is not included..?

image

**Review by tabVersion — APPROVED** [review]
thanks for the cleaning![birdbox](https://app.graphite.dev/macros/risingwavelabs)

**Comment by tabVersion** [comment]
> ```diff
> -   1000	1001	10465998	2015-07-15 00:00:22.005
> +   1000 1001 10465998 2015-07-15 00:00:22.005
> +   1000 1001 10465998 2015-07-15 00:00:22.005
> ```
> It seems it's `delete` not handled correctly.

Oops, I thought it was all about spacing. 
But the logic here does not involve handling delete

**Comment by xxchan** [comment]
I think it's sqllogictest (https://github.com/risinglightdb/sqllogictest-rs/pull/203) or the backwards-compat-tests's problem. But I don't know why only my PR triggered it 🤔

**Comment by xxchan** [comment]
I've reproduced this https://github.com/risingwavelabs/risingwave/pull/14354 🤣

**Review by st1page — APPROVED** [review]

**Pull request merged**